src/onyx/api.clj

```diff
@@ -218,22 +218,28 @@
   "Blocks until job-id has had all of its tasks completed or the job is killed.
    Returns true if the job completed successfully, false if the job was killed."
   ([peer-config job-id]
-     (await-job-completion peer-config job-id {:monitoring :no-op}))
+   (await-job-completion peer-config job-id {:monitoring :no-op}))
   ([peer-config job-id monitoring-config]
-     (let [job-id (validator/coerce-uuid job-id)
-           client (component/start (system/onyx-client peer-config monitoring-config))
-           ch (chan 100)]
-       (loop [replica (extensions/subscribe-to-log (:log client) ch)]
-         (let [entry (<!! ch)
-               new-replica (extensions/apply-log-entry entry replica)]
-           (cond (some #{job-id} (:completed-jobs new-replica))
-                 (do (component/stop client)
-                     true)
-                 (some #{job-id} (:killed-jobs new-replica))
-                 (do (component/stop client)
-                     false)
-                 :else 
-                 (recur new-replica)))))))
+   (await-job-completion peer-config job-id {:monitoring :no-op} nil))
+  ([peer-config job-id monitoring-config timeout-ms]
+   (-> (future
+         (let [job-id (validator/coerce-uuid job-id)
+               client (component/start (system/onyx-client peer-config monitoring-config))
+               ch (chan 100)]
+           (loop [replica (extensions/subscribe-to-log (:log client) ch)]
+             (let [entry (<!! ch)
```

Comment:
You can do the same trick you're doing below with a timeout channel to obviate using a `future`.